### PR TITLE
Update Script: Zigbee2MQTT dev branch install method from npm to pnpm

### LINKED ISSUE #698: Zigbee2mqtt dev branch issue
### Please verify that you have read and understood the guidelines.

yes

### A clear and concise description of the issue.

When running the default (non-alpine) version of the install script and choosing YES for the dev branch the installation of Z2M fails with an error.

The error shown is due to the use of `npm ci`, it will give an error regarding a missing lock file and does not continue after that, either it is missing `npm install` or it is because the dev branch recently changed to use `pnpm` instead which i think could also be the cause of this issue.
see: https://github.com/Koenkk/zigbee2mqtt/tree/dev?tab=readme-ov-file#developing

### What settings are you currently utilizing?

- [X] Default Settings
- [ ] Advanced Settings

### Which Linux distribution are you employing?

Debian 12

### If relevant, including screenshots or a code block can be helpful in clarifying the issue.

```
msg_info "Installing Zigbee2MQTT"
cd /opt/zigbee2mqtt
if [[ $DEV == "y" ]]; then
$STD git fetch origin dev:dev
$STD git checkout dev
$STD git pull
fi
$STD npm ci
msg_ok "Installed Zigbee2MQTT"
```

### Please provide detailed steps to reproduce the issue.

1. Copy the Z2M Default version of the install script to the proxmox shell
2. Proceed with the installation using the defaults
3. When asked to use the dev branch, answer `y`
4. Wait a little and it will show the `npm ci` error.

### PR DESCRIPTION
**Pull request by michelroegl-brunner**
> [!NOTE]
> We are meticulous when it comes to merging code into the main branch, so please understand that we may reject pull requests that do not meet the project's standards. It's never personal. Also, game-related scripts have a lower chance of being merged.

## Description

Changes the broken npm install for the dev branch to pnpm as stated by the maintainer of Zigbee2mqtt [KLICK](https://github.com/Koenkk/zigbee2mqtt/tree/dev?tab=readme-ov-file#developing)

Tested both, dev and no dev, all still working. 

I also tidied up the script :)

Fixes #698

## Type of change
Please check the relevant option(s):

- [X] Bug fix (non-breaking change that resolves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (a fix or feature that would cause existing functionality to change unexpectedly)
- [ ] New script (a fully functional and thoroughly tested script or set of scripts.)

## Prerequisites
The following efforts must be made for the PR to be considered. Please check when completed:
- [X] Self-review performed (I have reviewed my code, ensuring it follows established patterns and conventions)
- [X] Testing performed (I have tested my changes, ensuring everything works as expected)
- [ ] Documentation updated (I have updated any relevant documentation)

## Additional Information (optional)
Provide any additional context or screenshots about the feature or fix here.

## Related Pull Requests / Discussions

If there are other pull requests or discussions related to this change, please link them here:
- Related PR #

### CODE REVIEW

File: install/zigbee2mqtt-install.sh [hunk 1]
```diff
@@ -14,31 +14,31 @@ network_check
...
+$STD apt-get install -y curl \
+   sudo \
+   mc \
+   git \
+   make \
+   g++ \
+   gcc \
+   ca-certificates \
+   gnupg
+msg_ok "Installing Dependencies"
```
Comment (havardthom) [inline install/zigbee2mqtt-install.sh:26]:
```suggestion
msg_ok "Installed Dependencies"
```
What are you basing all these message changes on? They already look correct to me

File: install/zigbee2mqtt-install.sh [hunk 2]
```diff
@@ -47,15 +47,19 @@ else
   DEV="n"
 fi
 
-msg_info "Installing Zigbee2MQTT"
+msg_info "Setup Zigbee2MQTT"
```
Comment (havardthom) [inline install/zigbee2mqtt-install.sh:50]: change to this instead
```
if [[ $DEV == "y" ]]; then
  msg_info "Installing pnpm"
  $STD npm install -g pnpm
  msg_ok "Installed pnpm"

  msg_info "Installing Zigbee2MQTT"
  cd /opt/zigbee2mqtt
  $STD git fetch origin dev:dev
  $STD git checkout dev
  $STD git pull
  $STD pnpm install
  msg_ok "Installed Zigbee2MQTT"
else
  msg_info "Installing Zigbee2MQTT"
  cd /opt/zigbee2mqtt
  $STD npm ci
  msg_ok "Installed Zigbee2MQTT"
fi
```

**Comment by havardthom** [comment]
This change will not work in update function, since it uses `npm`.

I see that the change from `npm` to `pnpm` is part of a big Zigbee2MQTT v2.0.0 release which will be out on 3 January 2025. we should consider temporarily disabling choosing dev branch on install, until the official release. Then we can update the whole script to use new install and update method.

**Comment by michelroegl-brunner** [comment]
> This change will not work in update function, since it uses `npm`.
> 
> I see that the change from `npm` to `pnpm` is part of a big Zigbee2MQTT v2.0.0 release which will be out on 3 January 2025. we should consider temporarily disabling choosing dev branch on install, until the official release. Then we can update the whole script to use new install and update method.

Yes, I´ve also found that it is not done with installing the dev branch with pnmp, we would need to rewrite the whole script because npm is hardcoded in the service. It might be better to postpone. 

I´ve you want to keep going, i can rewrite it, otherwise please close the PR.

**Comment by MickLesk** [comment]
i think, only switch to new variant. 
Old Installations need to migrate himself. 

Theres a Breaking Change and here is the Announcement for this:
https://github.com/community-scripts/ProxmoxVE/discussions/660

**Comment by michelroegl-brunner** [comment]
Oh my, i completle missed that. I will close this PR and we will se when 2.0 is out.

**Pull request closed without merging**